integration-tests/admission/admission_suite_test.go

```diff
@@ -0,0 +1,183 @@
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/maya/integration-tests/artifacts"
+	"github.com/openebs/maya/integration-tests/kubernetes"
+	pod "github.com/openebs/maya/pkg/kubernetes/pod/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// auth plugins
+	//_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	// defaultTimeOut is the default time in seconds
+	// for Eventually block
+	defaultTimeOut int = 600
+	// defaultPollingInterval is the default polling
+	// time in seconds for the Eventually block
+	defaultPollingInterval int = 10
+	// minNodeCount is the minimum number of nodes
+	// neede to run this test
+	minNodeCount int = 1
+)
+
+func TestSource(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Admission")
+}
+
+func init() {
+	flag.Parse()
+}
+
+var _ = BeforeSuite(func() {
+	// Fetching the kube config path
+	configPath, err := kubernetes.GetConfigPath()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Setting the path in environemnt variable
+	err = os.Setenv(string(v1alpha1.KubeConfigEnvironmentKey), configPath)
+	Expect(err).ShouldNot(HaveOccurred())
+	// Getting clientset
+	cl, err := kubernetes.GetClientSet()
+	Expect(err).ShouldNot(HaveOccurred())
+
+	// Checking appropriate node numbers.
+	nodes, err := cl.CoreV1().Nodes().List(v1.ListOptions{})
+	Expect(nodes.Items).Should(HaveLen(minNodeCount))
+
+	// Fetching the openebs component artifacts
+	artifactsOpenEBS, errs := artifacts.GetArtifactsListUnstructuredFromFile(artifacts.OpenEBSArtifacts)
+	Expect(errs).Should(HaveLen(0))
+
+	// Installing the artifacts to kubernetes cluster
+	for _, artifact := range artifactsOpenEBS {
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(artifact),
+			artifact.GetNamespace(),
+		)
+		_, err := cu.Apply(artifact)
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+
+	// Check for maya-apiserver pod to get created and running
+	Eventually(func() int {
+		pods, err := pod.
+			KubeClient(pod.WithNamespace(string(artifacts.OpenebsNamespace))).
+			List(metav1.ListOptions{LabelSelector: string(artifacts.MayaAPIServerLabelSelector)})
+		Expect(err).ShouldNot(HaveOccurred())
+		return pod.ListBuilder().
+			WithAPIList(pods).
+			WithFilter(pod.IsRunning()).
+			List().
+			Len()
+	},
+		defaultTimeOut, defaultPollingInterval).
+		Should(Equal(1), "Maya-APIServer pod count should be 1")
+
+	// Check for provisioner pod to get created and running
+	Eventually(func() int {
+		pods, err := pod.
+			KubeClient(pod.WithNamespace(string(artifacts.OpenebsNamespace))).
+			List(metav1.ListOptions{LabelSelector: string(artifacts.OpenEBSProvisionerLabelSelector)})
+		Expect(err).ShouldNot(HaveOccurred())
+		return pod.ListBuilder().
+			WithAPIList(pods).
+			WithFilter(pod.IsRunning()).
+			List().
+			Len()
+	},
+		defaultTimeOut, defaultPollingInterval).
+		Should(Equal(1), "OpenEBS provisioner pod count should be 1")
```

Comment:
@AmitKumarDas @prateekpandey14 in IMO all suite_test.go we have code redundancy(checking for pod running state), to avoid this we can have a function and make the call to function when someone needs to check the pod status.
Eg: https://github.com/openebs/maya/blob/a4f62ab1f0b54b7c3072dd14f88c3ade4d585708/integration-tests/jiva/replicadeletion/jiva_replica_deletion_suit_test.go#L60